packages/bpk-component-floating-notification/src/BpkFloatingNotification.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React, { useEffect, useState, ReactElement } from 'react';
+import { CSSTransition } from 'react-transition-group';
+import BpkButton from 'bpk-component-button';
+import BpkText, { TEXT_STYLES } from 'bpk-component-text';
+import { cssModules } from 'bpk-react-utils';
+
+import STYLES from './BpkFloatingNotification.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export const THEME = {
+  light: 'light',
+  dark: 'dark',
+};
+
+export type Props = {
+  animateOnEnter: ?boolean,
+  animateOnExit: ?boolean,
+  className: ?string,
+  ctaText: ?string,
+  darkMode: ?boolean,
+  hideAfter: ?number,
+  icon: ?() => ReactElement,
+  onClick: ?() => void,
+  text: string,
+  theme: ?THEME,
```

Comment:
Can we call this `type` instead of theme? As these are two types rather than a theme, which we reserve this word to mean other things when we talk about themes/theming in Backpack